gen/src/generate/mod.rs

```diff
@@ -0,0 +1,38 @@
+mod char_property;
+
+mod idna;
+mod ucd;
+
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+
+/// The standard auto-generated source disclaimer.
+const PREAMBLE: &'static str = "// WARNING: Auto-generated by unic-gen. DO NOT EDIT MANUALLY!";
+
+pub fn generate_all(mut crates: Vec<String>) -> io::Result<()> {
+    crates.sort();
+
+    for krate in &crates {
+        let path = tables_path(krate);
+        if path.exists() {
+            fs::remove_dir_all(&path)?;
+        }
+        fs::create_dir_all(path)?;
+    }
+
+    idna::generate_all(crates.iter().filter(|x| x.starts_with("unic-idna")))?;
```

Comment:
I was assuming that each data source would correspond to one and only one major component.

I was specifying for the target component, but it would make just as much sense to specify by the data source. For now though, that difference is moot.